src/file.ts

```diff
@@ -2853,6 +2862,90 @@ class File extends ServiceObject<File> {
       .then(signedUrl => callback!(null, signedUrl), callback!);
   }
 
+  getFirebaseDownloadUrl(): Promise<GetFirebaseDownloadUrlResponse>;
+  getFirebaseDownloadUrl(callback: GetFirebaseDownloadUrlCallback): void;
+  /**
+   * @typedef {array} GetFirebaseDownloadUrlResponse
+   * @property {object} 0 The Firebase Storage download URL.
+   */
+  /**
+   * @callback GetFirebaseDownloadUrlCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} url The Firebase Storage download URL.
+   */
+  /**
+   * Get a Firebase download URL.
+   *
+   * Firebase download URLs use the `firebasestorage.googleapis.com` hostname, and have a static
+   * token attached to them. Unlike signed URLs, Firebase download URLs do not expire after a
+   * specific time period. Instead they remain valid until the underlying token is explicitly
+   * revoked. Developers can revoke these tokens from the Firebase Console or by directly modifying
+   * the object metadata.
+   *
+   * @param {GetFirebaseDownloadUrlCallback} [callback] Callback function.
+   * @returns {Promise<GetFirebaseDownloadUrlResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const myBucket = storage.bucket('my-bucket');
+   *
+   * const file = myBucket.file('my-file');
+   *
+   * //-
+   * // Generate a URL that allows temporary access to download your file.
+   * //-
+   * const request = require('request');
+   *
+   * file.getFirebaseDownloadUrl(function(err, url) {
+   *   if (err) {
+   *     console.error(err);
+   *     return;
+   *   }
+   *
+   *   // The file is now available to read from this URL.
+   *   request(url, function(err, resp) {
+   *     // resp.statusCode = 200
+   *   });
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * file.getFirebaseDownloadUrl().then(function(data) {
+   *   const url = data[0];
+   * });
+   */
+  getFirebaseDownloadUrl(
+    callback?: GetFirebaseDownloadUrlCallback
+  ): void | Promise<GetFirebaseDownloadUrlResponse> {
+    this.ensureDownloadToken()
+      .then(token => {
+        return `https://firebasestorage.googleapis.com/v0/b/${
```

Comment:
This shouldn't change your PR but FYI we're considering eventually building out an endpoint that users can call to disable download tokens on a per project or per bucket level.